src/main/java/org/openredstone/patch/DeathPotionsPatch.java

```diff
@@ -62,11 +101,13 @@ public void lingeringSplashPotionEvent(LingeringPotionSplashEvent event) {
 
     private boolean hasLegalEffects(List<PotionEffect> potionEffects) {
         for (PotionEffect effect : potionEffects) {
-            if (effect.getType().equals(PotionEffectType.HEAL) && (effect.getAmplifier() > maxAmplifier)) {
-                return false;
-            }
-            if (effect.getDuration() > maxDuration) {
-                return false;
+            if(!safeEffectsSet.contains(effect.getType())) {
```

Comment:
The logic could be less nested if you did `if (safeEffectsSet.contains(effect.getType())) continue;`, then have these `if` checks brought up one indentation.

Reply:
implemented this